common/src/main/java/com/wynntils/wrappedscreens/trademarket/TradeMarketSearchResultHolder.java

```diff
@@ -0,0 +1,370 @@
+        for (Int2ObjectMap.Entry<ItemStack> entry : currentPageEntries) {
+            if (ItemUtils.isItemEqual(entry.getValue(), clickedItem)) {
+                // Item found on the current page, click on it
+                WrappedScreenInfo wrappedScreenInfo = wrappedScreen.getWrappedScreenInfo();
+                ContainerUtils.clickOnSlot(
+                        entry.getIntKey(),
+                        wrappedScreenInfo.containerId(),
+                        GLFW.GLFW_MOUSE_BUTTON_LEFT,
+                        wrappedScreenInfo.containerMenu().getItems());
+
+                return;
+            }
+        }
+
+        // Assume the item is before the current page
+        int pageToLoad = 0;
+
+        // Check if the item is on a later page
+        for (int i = currentPage + 1; i < itemMap.size(); i++) {
+            boolean foundItem = false;
+
+            Int2ObjectOpenHashMap<ItemStack> itemsOnPage = itemMap.get(i);
+
+            for (ItemStack itemStack : itemsOnPage.values()) {
+                if (ItemUtils.isItemEqual(itemStack, clickedItem)) {
+                    // Item found on a later page, load the pages until the last one and click on it
+                    foundItem = true;
+                    pageToLoad = itemMap.size() - 1;
+                    break;
+                }
+            }
+
+            if (foundItem) break;
+        }
+
+        runOrQueueAction(new QueuedAction(PageLoadingMode.CLICK_ITEM, pageToLoad, clickedItem));
+    }
+
+    public void goBackToSearch() {
+        WrappedScreenInfo wrappedScreenInfo = wrappedScreen.getWrappedScreenInfo();
+        ContainerUtils.clickOnSlot(
+                BACK_TO_SEARCH_SLOT,
+                wrappedScreenInfo.containerId(),
+                GLFW.GLFW_MOUSE_BUTTON_LEFT,
+                wrappedScreenInfo.containerMenu().getItems());
+    }
+
+    public void loadNextPageBatch() {
+        if (allPagesLoaded) return;
+
+        runOrQueueAction(new QueuedAction(PageLoadingMode.LOAD_ITEMS, itemMap.size() - 1 + PAGE_BATCH_SIZE, null));
+    }
+
+    public void updateDisplayItems(ItemSearchQuery searchQuery) {
+        filteredItems.clear();
+
+        Stream<ItemStack> items = itemMap.values().stream()
+                .map(map -> map.values().toArray(new ItemStack[0]))
+                .flatMap(Stream::of);
+
+        List<ItemStack> matchingItems = Services.ItemFilter.filterAndSort(searchQuery, items.toList());
+
+        filteredItems.addAll(matchingItems);
+    }
+
+    public List<ItemStack> getFilteredItems() {
+        return filteredItems;
+    }
+
+    public int getPageLoadBatchSize() {
+        return PAGE_BATCH_SIZE;
+    }
+
+    private void handleSetItem(int slot, ItemStack itemStack) {
+        // We only care about the slots that are in the "search results" area
+        if (slot % 9 >= 7 || slot >= LAST_ITEM_SLOT) return;
+
+        // If we have the item count we expect on the page, we can load the next page
```

Comment:
Maybe move this to line 220? It took me a while to understand that this was not about the currentItems line below.

Reply:
Kind of outdated comment.. I will fix it.